### PR TITLE
PPTP-993 : General Partnership GRS Integration

### PR DESCRIPTION
**Pull request by RyanKellyHMRC**

### CODE REVIEW

**Review by tmc-ee — CHANGES_REQUESTED** [review]
I'm aware https://github.com/hmrc/plastic-packaging-tax-registration-frontend/pull/105 isn't merged yet, but, should we address [this comment](https://github.com/hmrc/plastic-packaging-tax-registration-frontend/pull/105#discussion_r673953808) as part of this PR or the original?

File: test/spec/PptTestData.scala [hunk 1]
```diff
@@ -16,7 +16,7 @@
 
 package spec
 
-import uk.gov.hmrc.plasticpackagingtax.registration.forms.{Address, OrgType}
+import uk.gov.hmrc.plasticpackagingtax.registration.forms.{OrgType}
```
[the remark this answers is not in the document]
Comment (RyanKellyHMRC) [reply test/spec/PptTestData.scala:19]: Imports are generally done semi-automatically by IntelliJ. I hadn't even noticed this.

File: app/uk/gov/hmrc/plasticpackagingtax/registration/connectors/GrsIntegration.scala [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+trait GrsEntityDetails
+
+trait GrsIntegration[
+  GrsCreateJourneyPayload <: GrsCreateJourneyRequest,
+  GrsResponse <: GrsEntityDetails
+] {
+  type RedirectUrl  = String
+  type GrsJourneyId = String
+
+  // TODO: consider implementing these methods here as they are very similar for all business entity types
```
Comment (PauloPortugal) [inline app/uk/gov/hmrc/plasticpackagingtax/registration/connectors/GrsIntegration.scala:32]: :+1:

File: conf/application.conf [hunk 3]
```diff
@@ -95,6 +95,12 @@ microservice {
       url = "http://localhost:9717"
     }
 
+	partnership-identification-frontend {
+      host = localhost
+      port = 9722
+      url = "http://localhost:9722"
+    }
+
```
Comment (PauloPortugal) [inline conf/application.conf:103]: You will need some additional QA/Staging/Base config

**Review by PauloPortugal — APPROVED** [review]
LGTM

**Review by tmc-ee — APPROVED** [review]

**Review by kiranprakashdigital — APPROVED** [review]

**Pull request merged**